tokio-udp/src/socket.rs

```diff
@@ -70,35 +77,49 @@ impl UdpSocket {
     ///
     /// # Return
     ///
-    /// On success, returns `Ok(Async::Ready(num_bytes_written))`.
+    /// On success, returns `Poll::Ready(Ok(num_bytes_written))`.
     ///
     /// If the socket is not ready for writing, the method returns
-    /// `Ok(Async::NotReady)` and arranges for the current task to receive a
+    /// `Poll::Pending` and arranges for the current task to receive a
     /// notification when the socket becomes writable.
     ///
     /// # Panics
     ///
     /// This function will panic if called from outside of a task context.
-    pub fn poll_send(&mut self, buf: &[u8]) -> Poll<usize, io::Error> {
-        try_ready!(self.io.poll_write_ready());
+    pub fn poll_send(
+        self: Pin<&mut Self>,
+        cx: &mut Context<'_>,
+        buf: &[u8],
+    ) -> Poll<io::Result<usize>> {
+        ready!(self.io.poll_write_ready(cx))?;
 
         match self.io.get_ref().send(buf) {
-            Ok(n) => Ok(n.into()),
             Err(ref e) if e.kind() == io::ErrorKind::WouldBlock => {
-                self.io.clear_write_ready()?;
-                Ok(Async::NotReady)
+                self.io.clear_write_ready(cx)?;
+                Poll::Pending
             }
-            Err(e) => Err(e),
+            x => Poll::Ready(x),
         }
     }
 
-    #[deprecated(since = "0.1.2", note = "use poll_recv instead")]
-    #[doc(hidden)]
-    pub fn recv(&mut self, buf: &mut [u8]) -> io::Result<usize> {
-        match self.poll_recv(buf)? {
-            Async::Ready(n) => Ok(n),
-            Async::NotReady => Err(io::ErrorKind::WouldBlock.into()),
-        }
+    /// Returns a future that receives a single datagram message on the socket from
```

Comment:
these docs seem wrong for recv, they seem like they are for recv_from?

Reply:
What's wrong? I just rephrased the doc of `poll_recv`, which is the same as `std::net::UdpSocket::recv`.

Reply:
Ahhhh I confused RecvFrom and Recv sorry 😄